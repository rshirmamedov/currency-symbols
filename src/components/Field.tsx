import { type InputHTMLAttributes, forwardRef } from "react";
import { Error } from "./Error";

type FieldProps = {
    inputId: string;
    label: string;
    error?: string;
} & InputHTMLAttributes<HTMLInputElement>;

export const Field = forwardRef<HTMLInputElement, FieldProps>(({ inputId, label, error, ...rest }, ref) => (
    <div data-testid={`${inputId}-field`} className="py-2">
        <label htmlFor={inputId} className="block text-sm font-medium leading-6 text-gray-900">
            {label}
        </label>
        <input
            id={inputId}
            className="w-full text-sm p-3 rounded outline-none border-2 focus:border-blue-500"
            {...rest}
            ref={ref}
        />
        {error && (
            <div className="pt-2">
                <Error message={error} />
            </div>
        )}
    </div>
));

Field.displayName = "Field";
