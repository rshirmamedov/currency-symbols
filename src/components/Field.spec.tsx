import { render } from "@testing-library/react";
import { Field } from "./Field";

describe("Field", () => {
    it("renders", () => {
        const { container } = render(<Field label="username" />);

        expect(container).toMatchSnapshot();
    });

    it("renders with error", () => {
        const { container } = render(<Field label="username" error="Username is required" />);

        expect(container).toMatchSnapshot();
    });
});
