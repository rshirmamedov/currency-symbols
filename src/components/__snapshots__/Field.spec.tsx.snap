// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders 1`] = `
<div>
  <div
    class="py-2"
    data-testid="undefined-field"
  >
    <label
      class="block text-sm font-medium leading-6 text-gray-900"
    >
      username
    </label>
    <input
      class="w-full text-sm p-3 rounded outline-none border-2 focus:border-blue-500"
    />
  </div>
</div>
`;

exports[`Field renders with error 1`] = `
<div>
  <div
    class="py-2"
    data-testid="undefined-field"
  >
    <label
      class="block text-sm font-medium leading-6 text-gray-900"
    >
      username
    </label>
    <input
      class="w-full text-sm p-3 rounded outline-none border-2 focus:border-blue-500"
    />
    <div
      class="pt-2"
    >
      <div
        class="text-sm text-red-600"
      >
        Username is required
      </div>
    </div>
  </div>
</div>
`;
